src/ModelContextProtocol.AspNetCore/AuthorizationMiddleware.cs

```diff
@@ -0,0 +1,65 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ModelContextProtocol.Protocol.Auth;
+using ModelContextProtocol.Server;
+using ModelContextProtocol.Utils.Json;
+using System.Text.Json;
+
+namespace ModelContextProtocol.AspNetCore;
+
+/// <summary>
+/// Middleware that handles authorization for MCP servers.
+/// </summary>
+internal class AuthorizationMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly ILogger<AuthorizationMiddleware> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AuthorizationMiddleware"/> class.
+    /// </summary>
+    /// <param name="next">The next middleware in the pipeline.</param>
+    /// <param name="logger">The logger factory.</param>
+    public AuthorizationMiddleware(RequestDelegate next, ILogger<AuthorizationMiddleware> logger)
+    {
+        _next = next ?? throw new ArgumentNullException(nameof(next));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    /// <summary>
+    /// Processes a request.
+    /// </summary>
+    /// <param name="context">The HTTP context.</param>
+    /// <param name="serverOptions">The MCP server options.</param>
+    /// <param name="authProvider">The authorization provider.</param>
+    /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+    public async Task InvokeAsync(
+        HttpContext context, 
+        IOptions<McpServerOptions> serverOptions,
+        IServerAuthorizationProvider? authProvider = null)
+    {
+        // Check if authorization is configured
+        if (authProvider == null)
+        {
+            // Authorization is not configured, proceed to the next middleware
+            await _next(context);
+            return;
+        }
+
+        // Handle the PRM document endpoint if not handled by the endpoint
+        if (context.Request.Path.StartsWithSegments("/.well-known/oauth-protected-resource") &&
```

Comment:
why is the metadata endpoint implemented as a middleware instead of as an app.Map call inside MapMcp?